WordPress/src/main/java/org/wordpress/android/viewmodel/StockMediaViewModel.java

```diff
@@ -0,0 +1,136 @@
+package org.wordpress.android.viewmodel;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.ViewModel;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+import org.wordpress.android.fluxc.Dispatcher;
+import org.wordpress.android.fluxc.generated.StockMediaActionBuilder;
+import org.wordpress.android.fluxc.model.StockMediaModel;
+import org.wordpress.android.fluxc.store.StockMediaStore;
+import org.wordpress.android.util.AppLog;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+public class StockMediaViewModel extends ViewModel {
+    private static final String KEY_SEARCH_QUERY = "KEY_SEARCH_QUERY";
+
+    private final Dispatcher mDispatcher;
+    private final StockMediaStore mStockMediaStore;
+
+    private String mSearchQuery;
+    private boolean mIsFetching;
+    private boolean mCanLoadMore;
+    private int mNextPage;
+
+    private final MutableLiveData<List<StockMediaModel>> mSearchResults;
+
+    @SuppressWarnings("WeakerAccess")
+    @Inject
+    public StockMediaViewModel(@NonNull Dispatcher dispatcher, @NonNull StockMediaStore stockMediaStore) {
+        super();
+
+        mDispatcher = dispatcher;
+        mStockMediaStore = stockMediaStore;
+        mDispatcher.register(this);
+
+        mSearchResults = new MutableLiveData<>();
+    }
+
+    @Override
+    protected void onCleared() {
+        super.onCleared();
+        mDispatcher.unregister(this);
+    }
+
+    public void writeToBundle(@NonNull Bundle outState) {
+        outState.putString(KEY_SEARCH_QUERY, mSearchQuery);
+    }
+
+    public void readFromBundle(@NonNull Bundle savedInstanceState) {
+        mSearchQuery = savedInstanceState.getString(KEY_SEARCH_QUERY);
+    }
+
+    public void fetchStockMedia(@Nullable String searchQuery, int page) {
+        if (TextUtils.isEmpty(searchQuery)) {
+            clearSearchResults();
+            return;
+        }
+
+        mIsFetching = true;
+
+        if (page == 1) {
+            clearSearchResults();
+        }
+
+        mSearchQuery = searchQuery;
+        AppLog.d(AppLog.T.MEDIA, "Fetching stock media page " + page);
+
+        StockMediaStore.FetchStockMediaListPayload payload =
+                new StockMediaStore.FetchStockMediaListPayload(searchQuery, page);
+        mDispatcher.dispatch(StockMediaActionBuilder.newFetchStockMediaAction(payload));
+    }
+
+    private void clearSearchResults() {
+        mSearchQuery = null;
+        mSearchResults.setValue(new ArrayList<StockMediaModel>());
```

Comment:
We should be using the `postValue` and worker threads as much as possible. The difference between `postValue` and `setValue` is that, `postValue` will always happen in the main thread even if it's called from a bg one. This will give us a way to use the `@WorkerThread` pretty much everywhere in the view model.